app/views/posts/index.html.erb

```diff
@@ -0,0 +1,13 @@
+<h1>Posts Index</h1>
+
+<% @posts.each do |post| %>
+
+<div>
+  <h4><%= post.title.upcase %></h4>
+  <p> <%= post.body %> </p>
+  <% if member_signed_in? %>
+    <p> By: <%= post.member.name %> </p>
+  <% end %>
+</div>
```

Comment:
`if..else` it's what we called logic.
It's not allowed in views.
Move that code in **views helpers**